pkg/codegen/docs/gen.go

```diff
@@ -2016,33 +2017,38 @@ func (dctx *docGenContext) getMod(
 	add bool,
 ) *modContext {
 	modName := pkg.TokenToModule(token)
-	mod, ok := modules[modName]
+	return dctx.getModByModName(pkg, tokens.ModuleName(modName), tokenPkg, modules, tool, add)
+}
+
+func (dctx *docGenContext) getModByModName(
+	pkg schema.PackageReference,
+	modName tokens.ModuleName,
+	tokenPkg schema.PackageReference,
+	modules map[string]*modContext,
+	tool string,
+	add bool,
+) *modContext {
+	mod, ok := modules[string(modName)]
 	if !ok {
 		mod = &modContext{
 			pkg:           pkg,
-			mod:           modName,
+			mod:           string(modName),
 			tool:          tool,
 			docGenContext: dctx,
 		}
 
 		if modName != "" && codegen.PkgEquals(tokenPkg, pkg) {
-			parentName := path.Dir(modName)
-			// If the parent name is blank, it means this is the package-level.
-			if parentName == "." || parentName == "" {
-				parentName = ":index:"
-			} else {
-				parentName = ":" + parentName + ":"
-			}
-			parent := dctx.getMod(pkg, parentName, tokenPkg, modules, tool, add)
-			if add {
+			parentName, hasParent := parentModule(modName)
+			if add && hasParent {
```

Comment:
`&& hasParent` has to be removed, otherwise the nav tree gets crippled because children don't get connected to parents.